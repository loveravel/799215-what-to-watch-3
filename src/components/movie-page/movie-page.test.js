import React from "react";
import renderer from "react-test-renderer";
import MoviePage from "./movie-page.jsx";

const movie = {
  id: 1,
  name: `Aviator`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  previewImage: `img/aviator.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  previewVideo: `https://`,
  video: `https://`,
  genre: `Drama / Epic Movie`,
  released: 2014,
  description: `Having received a small factory from his father,
    Howard Hughes turned it into a gigantic, fantastically profitable enterprise.
    Having become the owner of a huge film company, he shot the most expensive
    film of his time and won the hearts of the most beautiful Hollywood actresses.`,
  director: `Martin Scorsese`,
  starring: [
    `Leonardo DiCaprio`,
    `Cate Blanchett`,
    `Kate Beckinsale`,
    `John C. Reilly`,
    `Alec Baldwin`,
    `Alan Alda`,
    `Ian Holm`,
    `Danny Huston`,
    `Gwen Stefani`,
    `Jude Law`,
  ],
  runTime: 170,
  rating: 9.5,
  scoresCount: 15,
  isFavorite: false,
};

const reviews = [
  {
    id: 1,
    user: {
      id: 2,
      name: `Alex Muir`,
    },
    rating: 8.9,
    comment: `Not bad`,
    date: `2019-05-08T14:13:56.569Z`,
  },
  {
    id: 2,
    user: {
      id: 4,
      name: `Kate Muir`,
    },
    rating: 10,
    comment: `Wonderful! Wonderful! Wonderful! Wonderful! Wonderful!`,
    date: `2019-05-08T14:13:56.569Z`,
  },
];

it(`<MoviePage /> is rendered correctly`, () => {
  const tree = renderer
    .create(<MoviePage
      movie={movie}
      reviews={reviews}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
