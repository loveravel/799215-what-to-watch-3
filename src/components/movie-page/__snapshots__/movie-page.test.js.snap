// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePage /> is rendered correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__hero"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Aviator"
        src="img/img/bg-the-grand-budapest-hotel.jpg.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Aviator
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama / Epic Movie
          </span>
          <span
            className="movie-card__year"
          />
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="add-review.html"
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster movie-card__poster--big"
      >
        <img
          alt="The Grand Budapest Hotel poster"
          height="327"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item movie-nav__item--active"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Overview
              </a>
            </li>
            <li
              className="movie-nav__item"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Details
              </a>
            </li>
            <li
              className="movie-nav__item"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Reviews
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="movie-rating"
        >
          <div
            className="movie-rating__score"
          >
            9.5
          </div>
          <p
            className="movie-rating__meta"
          >
            <span
              className="movie-rating__level"
            >
              Very Good
            </span>
            <span
              className="movie-rating__count"
            >
              15
            </span>
          </p>
        </div>
        <div
          className="movie-card__text"
        >
          Having received a small factory from his father,
    Howard Hughes turned it into a gigantic, fantastically profitable enterprise.
    Having become the owner of a huge film company, he shot the most expensive
    film of his time and won the hearts of the most beautiful Hollywood actresses.
          <p
            className="movie-card__director"
          >
            <strong>
              Director: 
              Martin Scorsese
            </strong>
          </p>
          <p
            className="movie-card__starring"
          >
            <strong>
              Starring: 
              Leonardo DiCaprio, Cate Blanchett, Kate Beckinsale, John C. Reilly, Alec Baldwin, Alan Alda, Ian Holm, Danny Huston, Gwen Stefani, Jude Law
            </strong>
          </p>
        </div>
        <div
          className="movie-card__text movie-card__row"
        >
          <div
            className="movie-card__text-col"
          >
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Martin Scorsese
              </strong>
              <span
                className="movie-card__details-value"
              >
                Wes Andreson
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Starring
              </strong>
              <span
                className="movie-card__details-value"
              >
                Bill Murray, 
                <br />
                Edward Norton, 
                <br />
                Jude Law, 
                <br />
                Willem Dafoe, 
                <br />
                Saoirse Ronan, 
                <br />
                Tony Revoloru, 
                <br />
                Tilda Swinton, 
                <br />
                Tom Wilkinson, 
                <br />
                Owen Wilkinson, 
                <br />
                Adrien Brody, 
                <br />
                Ralph Fiennes, 
                <br />
                Jeff Goldblum
              </span>
            </p>
          </div>
          <div
            className="movie-card__text-col"
          >
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Run Time
              </strong>
              <span
                className="movie-card__details-value"
              >
                1h 39m
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Genre
              </strong>
              <span
                className="movie-card__details-value"
              >
                Comedy
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Released
              </strong>
              <span
                className="movie-card__details-value"
              >
                2014
              </span>
            </p>
          </div>
        </div>
        <div
          className="movie-card__reviews movie-card__row"
        >
          <div
            className="movie-card__reviews-col"
          >
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Not bad
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Alex Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    2019-05-08T14:13:56.569Z
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8.9
              </div>
            </div>
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Wonderful! Wonderful! Wonderful! Wonderful! Wonderful!
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    2019-05-08T14:13:56.569Z
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                10
              </div>
            </div>
          </div>
          <div
            className="movie-card__reviews-col"
          >
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Not bad
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Alex Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    2019-05-08T14:13:56.569Z
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                8.9
              </div>
            </div>
            <div
              className="review"
            >
              <blockquote
                className="review__quote"
              >
                <p
                  className="review__text"
                >
                  Wonderful! Wonderful! Wonderful! Wonderful! Wonderful!
                </p>
                <footer
                  className="review__details"
                >
                  <cite
                    className="review__author"
                  >
                    Kate Muir
                  </cite>
                  <time
                    className="review__date"
                    dateTime="2016-12-24"
                  >
                    2019-05-08T14:13:56.569Z
                  </time>
                </footer>
              </blockquote>
              <div
                className="review__rating"
              >
                10
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
